packages/react-reconciler/src/ReactFiberTracingMarkerComponent.new.js

```diff
@@ -96,6 +113,30 @@ export function processTransitionCallbacks(
         });
       }
 
+      const markerIncomplete = pendingTransitions.markerIncomplete;
+      const onMarkerIncomplete = callbacks.onMarkerIncomplete;
+      if (onMarkerIncomplete != null && markerIncomplete !== null) {
+        markerIncomplete.forEach(({transitions, deletions}, markerName) => {
+          transitions.forEach(transition => {
+            const filteredDeletions = [];
+            deletions.forEach(deletion => {
```

Comment:
Why does this `deletions` array exist? Why can't we schedule `onMarkerIncomplete` directly from the deletion effect?